src/mapping/GinkgoRadialBasisFctSolver.hpp

```diff
@@ -0,0 +1,533 @@
+
+enum SolverType {
+  CG,
+  GMRES,
+  MG
+};
+
+enum PreconditionerType {
+  Jacobi,
+  Cholesky,
+  Ilu,
+  None
+};
+
+const std::map<std::string, SolverType> solverTypeLookup{
+    {"cg-solver", SolverType::CG},
+    {"gmres-solver", SolverType::GMRES},
+    {"mg-solver", SolverType::MG}};
+
+const std::map<std::string, PreconditionerType> preconditionerTypeLookup{
+    {"jacobi-preconditioner", PreconditionerType::Jacobi},
+    {"cholesky-preconditioner", PreconditionerType::Cholesky},
+    {"ilu-preconditioner", PreconditionerType::Ilu},
+    {"no-preconditioner", PreconditionerType::None}};
+
+const std::map<std::string, std::function<std::shared_ptr<gko::Executor>()>> ginkgoExecutorLookup{{"reference-executor", [] { return gko::ReferenceExecutor::create(); }},
+                                                                                                  {"omp-executor", [] { return gko::OmpExecutor::create(); }},
+                                                                                                  {"cuda-executor", [] { return gko::CudaExecutor::create(0, gko::OmpExecutor::create(), true, gko::allocation_mode::device); }},
+                                                                                                  {"hip-executor", [] { return gko::HipExecutor::create(0, gko::OmpExecutor::create(), true); }}};
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+class GinkgoRadialBasisFctSolver {
+public:
+  GinkgoRadialBasisFctSolver() = default;
+
+  GinkgoRadialBasisFctSolver(const MappingConfiguration::GinkgoParameter &ginkgoParameter);
+
+  /// Assembles the system matrices and computes the decomposition of the interpolation matrix
+  template <typename IndexContainer>
+  GinkgoRadialBasisFctSolver(RADIAL_BASIS_FUNCTION_T basisFunction, const mesh::Mesh &inputMesh, const IndexContainer &inputIDs,
+                             const mesh::Mesh &outputMesh, const IndexContainer &outputIDs, std::vector<bool> deadAxis, Polynomial polynomial,
+                             const MappingConfiguration::GinkgoParameter &ginkgoParameter);
+
+  ~GinkgoRadialBasisFctSolver() = default;
+
+  GinkgoRadialBasisFctSolver(const GinkgoRadialBasisFctSolver &solver) = delete;
+  GinkgoRadialBasisFctSolver &operator=(const GinkgoRadialBasisFctSolver &solver) = delete;
+
+  /// Maps the given input data
+  Eigen::VectorXd solveConsistent(const Eigen::VectorXd &rhsValues, Polynomial polynomial);
+
+  /// Maps the given input data
+  Eigen::VectorXd solveConservative(const Eigen::VectorXd &inputData, Polynomial polynomial);
+
+  void clear();
+
+  // Access to the evaluation matrix (output x input)
+  const std::shared_ptr<GinkgoMatrix> getEvaluationMatrix() const;
+
+  std::shared_ptr<gko::Executor> getReferenceExecutor() const;
+
+private:
+  mutable precice::logging::Logger _log{"mapping::GinkgoRadialBasisFctSolver"};
+
+  std::shared_ptr<gko::Executor>        _deviceExecutor;
+  static std::shared_ptr<gko::Executor> _hostExecutor;
```

Comment:
Yes, it's the same. It is necessary to manage data transfers as it would be by using plain CUDA. At least I never found another way, and looking at the Ginkgo examples, they almost do the same.